src/containers/paint-editor.jsx

```diff
@@ -145,12 +170,58 @@ class PaintEditor extends React.Component {
     handleZoomReset () {
         resetZoom();
     }
+    setPaintEditor (paintEditor) {
+        this.paintEditor = paintEditor;
+    }
+    onMouseDown () {
+        if (this.props.isEyeDropping) {
+            const colorString = this.eyeDropper.colorString;
+            const callback = this.props.changeColorToEyeDropper;
+            
+            this.props.onDeactivateEyeDropper(this.props.previousMode);
+            this.stopEyeDroppingLoop();
+            if (!this.eyeDropper.hideLoupe) {
+                // If not hide loupe, that means the click is inside the canvas,
+                // so apply the new color
+                callback(colorString);
+            }
+            this.setState({colorInfo: null});
+        }
+    }
+    startEyeDroppingLoop () {
+        const canvas = this.paintEditor.getWrappedInstance().canvas;
```

Comment:
Instead of doing this, can you get the canvas directly by passing componentRef as a prop and having paint editor set the canvas? getWrappedInstance seems a bit hacky.

Reply:
I think I'm just going to use `document.getElement` rather than either, due to another bug I'm running into (a bug with refs that @paulkaplan mentions above)

Reply:
@fsih I changed it in a different way that prevents using `getElement` or `getWrappedInstance` – what do you think of the way it is now, with the canvas being a property rather than a state variable?